sonar-orchestrator/src/main/java/com/sonar/orchestrator/locator/GitHubImpl.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Orchestrator
+ * Copyright (C) 2011-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package com.sonar.orchestrator.locator;
+
+import com.eclipsesource.json.Json;
+import com.eclipsesource.json.JsonValue;
+import com.sonar.orchestrator.http.HttpCall;
+import com.sonar.orchestrator.http.HttpClientFactory;
+import okhttp3.HttpUrl;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.Optional;
+
+public class GitHubImpl implements GitHub {
+  private static final Logger LOG = LoggerFactory.getLogger(GitHubImpl.class);
+  private static String cachedVersion;
+  private final String baseUrl;
+
+  public GitHubImpl() {
+    this("https://api.github.com");
+  }
+
+  GitHubImpl(String baseUrl) {
+    this.baseUrl = baseUrl;
+  }
+
+  @Override
+  public synchronized Optional<String> getLatestScannerReleaseVersion() {
+    // GitHub API has a rate limit of 60 requests per hour for unauthenticated users.
+    // To avoid reaching that limit when running integration tests, the returned value needs to be cached.
+    if (cachedVersion != null) {
```

Comment:
Caching is a good idea anyway. Note that you could be more secure with API rate limit if you use the orchestrator GitHub token. If you look at [Licenses.java](https://github.com/SonarSource/orchestrator/blob/master/sonar-orchestrator/src/main/java/com/sonar/orchestrator/config/Licenses.java#L94) you'll see and example of its usage.
Orchestrator needs that to get licenses key.
But it's not a mandatory improvement I'd say, as 1/ caching will limit the number of hit per instance running, and 2/ the GitHub rate limit is likely IP based, so I guess different CI containers will be treated as different users.

Reply:
Thank you! That's very useful to know.

Since we have some work to be done that's waiting for this PR I would cut branches for now and merge without the authentication. If we notice that we will reach the limit I will add also the authentication.